src/all/pandachaika/src/eu/kanade/tachiyomi/extension/all/pandachaika/PandaChaika.kt

```diff
@@ -0,0 +1,303 @@
+        author = (groups ?: artists)
+        artist = artists
+        genre = listOfNotNull(male, female, others).joinToString().takeIf { it.isNotEmpty() }
+        description = buildString {
+            append("Uploader: ", uploader, "\n")
+            publishers?.let {
+                append("Publishers: ", it, "\n")
+            }
+            parodies?.let {
+                append("Parodies: ", it, "\n")
+            }
+            male?.let {
+                append("Male Tags: ", it, "\n")
+            }
+            female?.let {
+                append("Female Tags: ", it, "\n")
+            }
+            others?.let {
+                append("Other Tags: ", it, "\n\n")
+            }
+
+            title_jpn?.let { append("Japanese Title: ", it, "\n") }
+            append("Pages: ", filecount, "\n")
+
+            try {
+                append("Posted: ", dateReformat.format(Date(posted * 1000)), "\n")
+            } catch (_: Exception) {}
+        }
+        status = SManga.COMPLETED
+        update_strategy = UpdateStrategy.ONLY_FETCH_ONCE
+        initialized = true
+    }
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        return response.parseAs<Archive>().toSManga()
+    }
+
+    // Chapters
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val archive = response.parseAs<Archive>()
+
+        return listOf(
+            SChapter.create().apply {
+                name = "Chapter"
+                url = archive.download.substringBefore("/download/")
+                date_upload = archive.posted * 1000
+            },
+        )
+    }
+
+    override fun getMangaUrl(manga: SManga) = baseUrl + manga.url
+    override fun getChapterUrl(chapter: SChapter) = baseUrl + chapter.url
+
+    override fun imageRequest(page: Page): Request {
+        val remoteZip = page.imageUrl!!.substringAfter("and then this:").parseAs<RemoteZip>()
+        val filename = page.imageUrl!!.substringBefore("and then this:")
+
+        val uncompressedBytes = remoteZip.fetch(filename)
+        val imageUrl = Base64.encodeToString(uncompressedBytes, Base64.DEFAULT)
+
+        var type = filename.substringAfterLast('.').lowercase()
+        type = if (type == "jpg") "jpeg" else type
+
+        return GET("https://127.0.0.1/?images/$type;base64,$imageUrl", headers)
+    }
+
+    // Pages
+    override fun pageListParse(response: Response): List<Page> {
+        fun List<String>.sort() = this.sortedBy { it.filter(Char::isDigit).toIntOrNull() }
+        val archive = response.parseAs<Archive>()
+        val url = "$baseUrl${archive.download}".toHttpUrl()
+        val remoteZip = RemoteZipPointer(url, additionalHeaders = headers).populate()
+        val fileListing = remoteZip.files().sort()
+
+        val file = remoteZip.toJson()
+        return fileListing.mapIndexed { index, _ ->
+            Page(index, imageUrl = fileListing[index] + "and then this:" + file)
```

Comment:
```suggestion
        return fileListing.mapIndexed { index, img ->
            Page(index, imageUrl = img + "#" + file)
```
this will add to the url fragment which can easily be checked and accessed in interceptor